Sources/DataPipeline/DataPipelineImplementation.swift

```diff
@@ -76,11 +81,49 @@ class DataPipelineImplementation: DataPipelineInstance {
     var profileAttributes: [String: Any] {
         get { analytics.traits() ?? [:] }
         set {
-            let userId = analytics.userId ?? analytics.anonymousId
-            analytics.identify(userId: userId, traits: newValue)
+            let userId = analytics.userId
+            guard let userId = userId else {
```

Comment:
I have skipped empty/blank checks to match `analytics`

Reply:
I think this is OK. If `analytics.userId` is empty or blank and we don't want that, it's too late at this point in time, setting profile attributes. We should have prevented `analytics.userId` being set with an empty/blank id in `identify`.